### PR TITLE
Add onRowMouseEnter and onRowMouseLeave event handlers

### PR DESCRIPTION
**Pull request by gsimko**
This PR picks up where the abandoned https://github.com/jbetancur/react-data-table-component/pull/807 was left and adds both the enter & leave events so that hovering can be properly detected.

### CODE REVIEW

**Comment by gsimko** [comment]
@jbetancur can you please take a look at this PR?

File: .vscode/settings.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
 		"editor.defaultFormatter": "esbenp.prettier-vscode"
 	},
 	"[typescriptreact]": {
-		"editor.defaultFormatter": "esbenp.prettier-vscode"
+		"editor.defaultFormatter": "vscode.typescript-language-features"
```
Comment (jbetancur) [inline .vscode/settings.json:35]: Is this needed due to an update from prettier?

Comment (gsimko) [reply .vscode/settings.json]: No, it was an accidental change by vscode. I've reverted it.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-data-table-component",
-  "version": "7.4.7",
+  "version": "7.4.8",
```
Comment (jbetancur) [inline package.json:3]: This is a new feature so it should be 7.5.0

Comment (jbetancur) [reply package.json:3]: One last comment and this should be GTG

Comment (gsimko) [reply package.json]: done

**Pull request merged**